marshaler.go

```diff
@@ -136,19 +150,18 @@ func (enc *Encoder) encode(b []byte, ctx encoderCtx, v reflect.Value) ([]byte, e
 		if v.IsNil() {
 			return nil, errNilInterface
 		}
+
```

Comment:
I'm really doubting the improvement in readability of new lines before return inside switch statements. No chance we can disable it?

Reply:
it's the https://github.com/ssgreg/nlreturn linter. There is no config for this unfortunately.
imho, it actually improves readability (as someone who was reading the code for the 1st time, it felt helpful), but that's definitely extremely subjective.

In any case, I think the only alternative here is: remove nlreturn altogether or keep it, I let you decide!

Reply:
Cool, let's keep it and see if I get used to it :)